Sources/PackageModel/Manifest/PackageConditionDescription.swift

```diff
@@ -52,6 +64,55 @@ struct PackageConditionWrapper: Codable, Equatable, Hashable {
             fatalError("unknown condition \(condition)")
         }
     }
+
+    init(_ condition: PackageCondition) {
+        switch condition {
+        case let .platforms(platforms):
+            self.platform = platforms
+        case let .configuration(configuration):
+            self.config = configuration
+        }
+    }
+}
+
+/// One of possible conditions used in package manifests to restrict targets from being built for certain platforms or
+/// build configurations.
+public enum PackageCondition: Hashable {
+    case platforms(PlatformsCondition)
+    case configuration(ConfigurationCondition)
+
+    public func satisfies(_ environment: BuildEnvironment) -> Bool {
+        switch self {
+        case .configuration(let configuration):
+            return configuration.satisfies(environment)
+        case .platforms(let platforms):
+            return platforms.satisfies(environment)
+        }
+    }
+
+    public var platformsCondition: PlatformsCondition? {
```

Comment:
are these really required?

Reply:
These are used below in `.compactMap(\.configurationCondition)` replacing previous `.compactMap({ $0 as? ConfigurationCondition })`.